### PR TITLE
feat(devices): differentiation between out-of-band and in-band power actions

### PR DESCRIPTION
**Pull request by intelkrishi**
## PR Checklist

- [x] Unit Tests have been added for new changes
- [x] API tests have been updated if applicable
- [x] All commented code has been removed
- [x] If you've added a dependency, you've ensured license is compatible with Apache 2.0 and clearly outlined the added dependency.

### CODE REVIEW

File: src/app/devices/device-detail/device-detail.component.html [hunk 1]
```diff
@@ -1,34 +1,141 @@
...
+                    <mat-card-title>
+                        Out-of-Band Power Actions
+                    </mat-card-title>
+                    <mat-card-subtitle>Actions occur at the hardware level. Use these
+                        with caution as they are akin to disconnecting the power cable from the device.</mat-card-subtitle>
+
+                </mat-card-header>
+                <mat-card-content>
+                    <!--
+                    <button *ngFor="let option of powerOptions" mat-flat-button>
```
Comment (rsdmike) [inline src/app/devices/device-detail/device-detail.component.html:50]: i was thinking we could keep these buttons data driven. Is there a particular reason why you didn't want to do that?

File: src/app/devices/device-detail/device-detail.component.scss [hunk 2]
```diff
@@ -37,4 +37,24 @@ p{
   span{
     color:black
   }
+}
+
+.oobpowerActions {
```
Comment (rsdmike) [inline src/app/devices/device-detail/device-detail.component.scss:42]: ideally we aren't adding custom css if we can avoid it. I think we can accomplish this by using the the flex tags provided with angular flex. Additionally, these aren't wrapping like they should when the window is smaller.
image

**Review by rsdmike — CHANGES_REQUESTED** [review]
On the right track, have you had a chance to review the UI changes with either @Kimberly-Davis or @bwendlandt-intel ?

File: cypress/integration/device/device-details.spec.ts [hunk 3]
```diff
@@ -55,6 +60,34 @@ describe('Test device details page', () => {
     cy.wait('@get-auditlog').its('response.statusCode').should('eq', 200)
     cy.wait('@get-features').its('response.statusCode').should('eq', 200)
 
+    // Out-of-band Power Actions
+    cy.contains('Power On').click()
```
Comment (rsdmike) [inline cypress/integration/device/device-details.spec.ts:64]: these look like good candidates for a for loop

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: cypress/integration/device/device-details.spec.ts (hunk 3 above)
Comment (intelkrishi) [reply cypress/integration/device/device-details.spec.ts]: fixed

**Review by rsdmike — APPROVED** [review]

**Pull request merged**